libgadget/density.c

```diff
@@ -284,10 +285,11 @@ density_reduce(int place, TreeWalkResultDensity * remote, enum TreeWalkReduceMod
         TREEWALK_REDUCE(SPHP(place).GradRho[2], remote->GradRho[2]);
 #endif
 
-#ifdef DENSITY_INDEPENDENT_SPH
-        TREEWALK_REDUCE(SPHP(place).EgyWtDensity, remote->EgyRho);
-        TREEWALK_REDUCE(SPHP(place).DhsmlEgyDensityFactor, remote->DhsmlEgyDensity);
-#endif
+        /*Only used for density independent SPH*/
+        if(All.DensityIndependentSphOn) {
```

Comment:
Shall we zero out the values if it is not On? We haven't been doing that on other branches, but I do worry about undefined values tainting valgrind reports very late down stream.

Reply:
hmm, I think not. EgyWtDensity is initialised to -1 in init.c and should never be written to, so it is already a canary. Both DhsmlDensityFactor variables should be explicitly initialized to zero though.

Reply:
Both these variables are already initialized, but in a non-obvious way.